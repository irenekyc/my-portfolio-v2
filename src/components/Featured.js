import React from 'react'

const Featured = ()=>{
    return(
        <section className="featured-session">
            <div className="featured-inner">
                <div className="featured-heading">
                    <i className="fab fa-react"></i>
                    <i className="fab fa-node-js"></i>

                </div>
                <div className="featured-title">
                    <h5> <span>FEATURED </span></h5>
                    <h1>MERN project - This portfolio</h1>
                </div>
                
                <div className="featured-image">
                    <img src="images/my-portfolio.png" alt="portfolio" />
                </div>
                <div className="featured-content">
                    <h4> MERN stands for MongoDB, Express, React and Node js.  </h4>
                    <p> This portfolio is designed with MERN stack. MongoDB is mainly for storing and updating my project database.
                        I use express and node to enable backend api routing to fetch data. React play an essential role to connect my web design and backend environment.
                        Hope you enjoy my site!
                    </p>
                    <p> Next step, the admin login panel and updating projects form are under construction. This will apply CRUD method</p>
                </div>
            </div>
        </section>

    )
}
export default Featured