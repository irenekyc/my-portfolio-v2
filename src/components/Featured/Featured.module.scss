.ikFeaturedSession {
  width: 100vw;
  padding: 2rem 0;
  &__inner {
    width: 90%;
    max-width: 700px;
    margin-right: auto;
    margin-left: auto;
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-orient: vertical;
    -webkit-box-direction: normal;
    -ms-flex-direction: column;
    flex-direction: column;
    padding: 1rem;
    border: 2px solid #b7985b;
    border-radius: 15px;
    -webkit-box-shadow: 5px 5px 10px var(#b7985b, 0.8) -5px 5px 10px var(#b7985b, 0.8);
    box-shadow: 5px 5px 10px var(#b7985b, 0.8) -5px 5px 10px var(#b7985b, 0.8);

    &__title {
      margin: 0.5rem 0;
      h5 {
        padding: 0.6rem 0;
        span {
          padding: 0.5rem;
          font-size: 1rem;
          background: #b50800;
          border-radius: 5px;
        }
      }
    }

    &__heading {
      padding: 0.5rem 0;
      font-size: 2rem;
    }
    &__image {
      height: 150px;
      padding: 0.5rem 0;
      img {
        height: 100%;
        width: auto;
      }
    }
    &__content {
      padding: 0.5rem 0;
      line-height: 1.8;
    }
  }
}
