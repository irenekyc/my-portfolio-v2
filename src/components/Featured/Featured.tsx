import React from "react";
import styles from "./Featured.module.scss";

const Featured = () => {
  return (
    <section className={styles.ikFeaturedSession}>
      <div className={styles.ikFeaturedSession__inner}>
        <div className="featured-heading">
          <i className="fab fa-react"></i>
        </div>
        <div className={styles.ikFeaturedSession__inner__title}>
          <h5>
            {" "}
            <span>FEATURED </span>
          </h5>
          <h2 className={styles.ikFeaturedSession__inner__heading}>
            Portfolio Showcase V2
          </h2>
        </div>

        <div className={styles.ikFeaturedSession__inner__image}>
          <img src="images/my-portfolio.png" alt="portfolio" />
        </div>
        <div className={styles.ikFeaturedSession__inner__content}>
          <p>Use Google sheet as CMS for portfolio showcase</p>
          <p>
            Welcome to mysite. This site is built with React and connect Google
            sheets via google-spreadsheet package. Styles are built at
            react-bootstrap barebone. Additionally, I upgrade the project using
            Typescript.
          </p>
          <p>
            Next Step, I will create an admin panel to add new project data to
            write to google sheets
          </p>
        </div>
      </div>
    </section>
  );
};
export default Featured;
