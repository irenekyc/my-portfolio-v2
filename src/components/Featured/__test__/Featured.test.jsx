import React from "react";
import Featured from "../index";
import { render, screen } from "@testing-library/react";

it("Render Featured Section", () => {
  render(<Featured />);
  const featuredSection = screen.getByTestId("featured-portfolio");
  expect(featuredSection).toBeInTheDocument();
  const featuredSectionTitle = screen.getByTestId("featured-portfolio-title");
  expect(featuredSectionTitle).toBeInTheDocument();
  const featuredSectionimage = screen.getByTestId("featured-portfolio-image");
  expect(featuredSectionimage).toBeInTheDocument();
  const featuredSectioncontent = screen.getByTestId(
    "featured-portfolio-content"
  );
  expect(featuredSectioncontent).toBeInTheDocument();
});
