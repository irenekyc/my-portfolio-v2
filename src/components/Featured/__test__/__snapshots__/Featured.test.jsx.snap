// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Featured Section 1`] = `
<section
  class="ikFeaturedSession"
  data-testid="featured-portfolio"
  id="featured-section"
>
  <div
    class="ikFeaturedSession__inner"
  >
    <div
      class="featured-heading"
    >
      <i
        class="fab fa-react"
      />
    </div>
    <div
      class="ikFeaturedSession__inner__title"
    >
      <h5>
         
        <span>
          FEATURED 
        </span>
      </h5>
      <h2
        class="ikFeaturedSession__inner__heading"
        data-testid="featured-portfolio-title"
      >
        Portfolio Showcase V2
      </h2>
    </div>
    <div
      class="ikFeaturedSession__inner__image"
    >
      <img
        alt="portfolio"
        data-testid="featured-portfolio-image"
        src="images/my-portfolio.png"
      />
    </div>
    <div
      class="ikFeaturedSession__inner__content"
      data-testid="featured-portfolio-content"
    >
      <p>
        Use Google sheet as CMS for portfolio showcase
      </p>
      <p>
        Welcome to mysite. This site is built with React and connect Google sheets via google-spreadsheet package. Styles are built at react-bootstrap barebone. Additionally, I upgrade the project using Typescript.
      </p>
      <p>
        Next Step, I will create an admin panel to add new project data to write to google sheets
      </p>
    </div>
  </div>
</section>
`;
